### PR TITLE
build: Micronaut Framework to 3.5.4

### PR DESCRIPTION
**Pull request by sdelamo**
- Micronaut Framework to 3.5.4
- Micronaut Test to 3.4.0
- Groovy to 3.0.12
- Spock 2.1

It seems @burtbeckwith had a stab at fixing the build.

https://github.com/micronaut-projects/micronaut-jms/pull/263

### CODE REVIEW

**Comment by wetted** [comment]
This is a work in progress, with some tests still failing.

**Comment by wetted** [comment]
this should satisfy the task for https://github.com/micronaut-projects/micronaut-core/issues/7720

File: buildSrc/src/main/groovy/io.micronaut.build.internal.jms-module.gradle [hunk 1]
```diff
@@ -5,4 +5,12 @@ plugins {
 
 dependencies {
     compileOnly mn.jsr305 // for "warning: unknown enum constant When.MAYBE"
+    runtimeOnly("javax.json:javax.json-api:1.1.4")
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.jms-module.gradle:8]: Should this be in the catalog?

File: buildSrc/src/main/groovy/io.micronaut.build.internal.jms-tests.gradle [hunk 2]
```diff
@@ -18,6 +18,9 @@ dependencies {
     testImplementation "io.micronaut.test:micronaut-test-core:$micronautTestVersion"
     testImplementation mn.micronaut.http.client
     testImplementation libs.testcontainers.spock
+
+    // without this :tests:tasks-activemq-artemis TasksSpec fails with ClassNotFoundException for javax.json.JsonValue
+    runtimeOnly 'javax.json:javax.json-api:1.1.4'
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.jms-tests.gradle:23]: Should this be in the catalog?

File: gradle.properties [hunk 3]
```diff
@@ -13,11 +13,12 @@ githubSlug=micronaut-projects/micronaut-jms
 developers=Graeme Rocher
 
 githubBranch=master
-githubCoreBranch=3.3.x
+githubCoreBranch=3.5.x
```
Comment (timyates) [inline gradle.properties:16]: Should this be 3.7.x?

**Review by timyates — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 3 above)
Comment (sdelamo) [reply gradle.properties:16]: yes

**Pull request merged**